src/commons/secret.rs

```diff
@@ -0,0 +1,36 @@
+use std::fmt::Display;
+
+use k8s_openapi::api::core::v1::Secret;
+use kube::runtime::reflector::ObjectRef;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+// Redefine SecretReference instead of reusing k8s-openapi's, in order to make name/namespace mandatory.
```

Comment:
Should be a doccomment. We should also clarify that this is _must_ only be used in cluster-scoped objects. Namespaced objects _must not_ be able to define cross-namespace secret references.

Reply:
I basically copied this straight from https://github.com/stackabletech/secret-operator/blob/main/rust/crd-utils/src/lib.rs#L12-L37 but I can adjust the comment as you requested.

Reply:
Added in 17a6aef

Reply:
This isn't about them being mandatory, an optional `namespace` field would still be problematic in a namespaced object.

Reply:
Adjusted in 694e337. Is this better now?